Lesson4.1/src/com/home/model/Computer.java

```diff
@@ -0,0 +1,97 @@
+package com.home.model;
+
+import com.home.service.Reader;
+
+import java.util.Random;
+
+public class Computer {
+
+    private String cpu;
+    private String ram;
+    private String hardDisk;
+    private int lifeCycle;
+
+    public Computer(String cpu, String ram, String hardDisk, int lifeCycle) {
+        this.cpu = cpu;
+        this.ram = ram;
+        this.hardDisk = hardDisk;
+        this.lifeCycle = lifeCycle;
+    }
+
+    public void info() {
+        System.out.println("Model of CPU: " + cpu);
+        System.out.println("Model of RAM: " + ram);
+        System.out.println("Model of hard disk: " + hardDisk);
+        System.out.println("Number of life cycle: " + lifeCycle);
+    }
+
+    public int turnOn(int lifeCycle) {
+        //int cycle = lifeCycle;
+        System.out.println("PC is turning on...");
+        if (getRandomNumber()) {
+            lifeCycle = turnOff(lifeCycle);
+            if (lifeCycle == 0) {
+                System.out.println("PC has burned down! Out of life cycles! ");
+            }
+        } else {
+            System.out.println("PC has burned down!!!");
+            lifeCycle = 0;
+        }
+        return lifeCycle;
+
+    }
+
+    public int turnOff(int lifeCycle) {
+        System.out.println("PC is turning off...");
+        int cycle = lifeCycle;
+        if (getRandomNumber()) {
+            System.out.println("PC is OFF!");
+            cycle--;
```

Comment:
so it seems like you do cycle value negative here, and in that case if statement on line 33 would not work

Reply:
In my case in Main.java value lifeCycles always > 0, so in that way case-- never be negative. But in another way, yes i understand it could be negative. So, my question is: how do we check methods turnOn and turnOff in Main? I  thought that it should be in while cycle.

Reply:
you need to track value of cycle variable here, in this class. And if it ever becomes 0 - it means that PC is dead